src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -337,6 +320,11 @@ internal JsonTypeInfo(Type type, JsonConverter converter, Type runtimeType, Json
                     Debug.Fail($"Unexpected class type: {PropertyInfoForTypeInfo.ConverterStrategy}");
                     throw new InvalidOperationException();
             }
+
+            // These two method overrides are expected to perform
+            // orthogonal changes, so we can invoke them both safely.
+            converter.ConfigureJsonTypeInfo(this, options);
+            converter.ConfigureJsonTypeInfoUsingReflection(this, options);
```

Comment:
why does the converter need to be configured twice here? (possibly this is correct but feels one of the method should call the other in that case)